packages/flutter_tools/test/src/process.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:process/process.dart';
+
+/// Matches only strings that a shell will always parse as a single literal word.
+///
+/// Some strings that a shell would in fact accept as a single literal word
+/// will not match this pattern.  This pattern should only be used when
+/// an error in that direction would be merely cosmetic.
+final RegExp _definitelyShellLiteralWordRegExp = RegExp(r'^[a-zA-Z0-9./,_-]+$');
+
+/// A string which a shell would parse as the given string value.
+///
+/// This method makes some effort to return the value unchanged,
+/// for the sake of a clean appearance, when doing so meets the requirements.
+/// For example, `shellEscapeString("asdf") == "asdf"`.
+///
+/// See also [shellEscapeCommand], for operating on a whole command line.
+String shellEscapeString(String value) {
+  if (_definitelyShellLiteralWordRegExp.hasMatch(value)) {
+    return value;
+  }
+  return "'${value.replaceAll("'", r"'\''")}'";
+}
+
+/// A string which a shell would parse as the given command.
+///
+/// Useful for printing a command unambiguously, or for printing
+/// a command the user might want to copy-paste and run.
+///
+/// This method makes some effort to print the command's elements
+/// verbatim, for the sake of a clean appearance, where possible.
+/// For example, `shellEscapeCommand(['git', 'commit', '-am', 'a commit'])`
+/// returns `git commit -am 'a commit'`.
+///
+/// See also [shellEscapeString], for operating on an individual
+/// argument of a command.
+String shellEscapeCommand(List<String> command) {
+  return command.map(shellEscapeString).join(' ');
+}
+
+extension ProcessManagerExtension on ProcessManager {
```

Comment:
Can you make these non-extension helper methods? https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension